app/src/test/java/org/mozilla/tv/firefox/helpers/RxSchedulerBeforeClass.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.helpers
+
+import io.reactivex.Scheduler
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.functions.Function
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.schedulers.TestScheduler
+import org.junit.BeforeClass
+import org.junit.rules.ExternalResource
+import java.util.concurrent.Callable
+
+/**
+ * Used to force all tested Rx code to execute synchronously.
+ *
+ * IMPORTANT NOTE: **must** be called from [BeforeClass]
+ */
+fun forceRxSynchronous() {
+    setRxScheduler(Schedulers.trampoline())
+}
+
+/**
+ * Used to force all tested Rx code that changes threads to operate on the same
+ * [TestScheduler].
+ *
+ * IMPORTANT NOTE: **must** be called from [BeforeClass]
+ *
+ * @return the [TestScheduler] to which all observeOn and subscribeOn calls will
+ * be forwarded.
+ */
+fun forceRxTestScheduler(): TestScheduler {
+    val testScheduler = TestScheduler()
+    setRxScheduler(testScheduler)
+    return testScheduler
+}
+
+private fun setRxScheduler(scheduleTo: Scheduler) {
+    val initHandler = Function<Callable<Scheduler>, Scheduler> { scheduleTo }
+    val setHandler = Function<Scheduler, Scheduler> { scheduleTo }
+
+    RxJavaPlugins.setInitIoSchedulerHandler(initHandler)
```

Comment:
In the original code, we only used two of these: why are all of these necessary?

Reply:
`CursorModelTest` only uses the Android main thread and the computation scheduler, but other code could use all the rest of these.  I've personally only ever wanted all of my schedulers to operate the same way during tests, and if we ever run into a different use case we can always just not use this.